slo_compliance.py

```diff
@@ -0,0 +1,111 @@
+
+no = '-'
+Types = namedtuple('Label',
+                   ['bug', 'feature', 'question', 'cleanup', 'process'])
+types = Types('label:"type: bug" ',
+              'label:"type: feature request" ',
+              'label:"type: question" ',
+              'label:"type: cleanup" ',
+              'label:"type: process" ')
+
+Priority = namedtuple('Priority', ['p0', 'p1', 'p2'])
+priority = Priority('label:"priority: p0" ',
+                    'label:"priority: p1" ',
+                    'label:"priority: p2" ')
+
+
+def updated(days):
+    return 'updated:<={}'.format(today-timedelta(days))
+
+
+def created(days):
+    return 'created:<={}'.format(today-timedelta(days))
+
+
+def query(url, params, name):
+    r = requests.get(url, params=params)
+    soup = BeautifulSoup(r.text, 'html.parser')
+    number = int(soup
+                 .find('a', class_='btn-link selected')
+                 .contents[2]
+                 .strip()[:-5])
+    items = soup.find_all('span', class_='opened-by')
+    print(r.url)
+    print(name, number)
+    print([int(item.contents[0].strip().split("\n")[0][1:]) for item in items])
+
+
+issues = {'issues_with_no_type': (no+types.bug
+                                  + no+types.feature
+                                  + no+types.question
+                                  + no+types.cleanup
+                                  + no+types.process),
+          'questions_with_no_assignee': 'no:assignee ',
+          'bugs_with_no_assignee': types.bug + 'no:assignee',
+          'bugs_with_no_priority': (types.bug
+                                    + no+priority.p0
+                                    + no+priority.p1
+                                    + no+priority.p2),
+          'bugs_with_priority_p0': types.bug + priority.p0,
+          'bugs_with_priority_p1': types.bug + priority.p1,
+          'bugs_with_priority_p2': types.bug + priority.p2,
+          'questions_outside_response_slo': types.question,
+          'p0_bugs_outside_response_slo': (types.bug
+                                           + priority.p0
+                                           + updated(1)),
+          'p0_bugs_outside_closure_slo': (types.bug
+                                          + priority.p0
+                                          + created(5)),
+          'p1_bugs_outside_response_slo': (types.bug
+                                           + priority.p1
+                                           + updated(5)),
+          'p1_bugs_outside_closure_slo': (types.bug
+                                          + priority.p1
+                                          + created(42)),
+          'p2_bugs_outside_response_slo': (types.bug
+                                           + priority.p2
+                                           + updated(120)),
+          'p2_bugs_outside_closure_slo': (types.bug
+                                          + priority.p2
+                                          + created(12)),
```

Comment:
I don't think this makes sense (12 days closure vs. 120 days response).  Use constants rather than "magic numbers", and the problem would be more obvious.

Reply:
Changed 12 to 120.

I *think* I will leave the numbers here rather than putting it in constants higher up unless you feel strongly about it?  They are fairly separate issues and it is easier to see that way.  Unless we use these numbers elsewhere, then I will consider refactoring them out.  But otherwise you will just see a variable that you have to scroll above for.